rust/kernel/sync/lock/spinlock.rs

```diff
@@ -121,13 +123,24 @@ unsafe impl super::Backend for SpinLockBackend {
             None => unsafe { bindings::spin_unlock(ptr) },
         }
     }
+
+    unsafe fn relock(ptr: *mut Self::State, state: &mut Self::GuardState) {
+        let _ = match state {
+            // SAFETY: The safety requiments of this function ensure that `ptr` has been
+            // initialised.
+            None => unsafe { Self::lock(ptr) },
+            // SAFETY: The safety requiments of this function ensure that `ptr` has been
+            // initialised.
+            Some(_) => unsafe { Self::lock_irqsave(ptr) },
+        };
+    }
 }
 
 // SAFETY: The underlying kernel `spinlock_t` object ensures mutual exclusion. We use the `irqsave`
 // variant of the C lock acquisition functions to disable interrupts and retrieve the original
 // interrupt state, and the `irqrestore` variant of the lock release functions to restore the state
 // in `unlock` -- we use the guard context to determine which method was used to acquire the lock.
-unsafe impl super::IrqSaveBackend for SpinLockBackend {
+unsafe impl IrqSaveBackend for SpinLockBackend {
```

Comment:
nit: you could maybe import it on the first commit, then you could avoid having the change in this commit.

Reply:
I'm trying to avoid importing things that are only used once in a place with no space constraints.

It may not be clear, but I imported it here because I use `Self::lock_irqsave` from `relock` -- it would require an explicit `<Self as IrqSaveBackend>::lock_irqsave` otherwise (which would the second use of `IrqSaveBackend`.

Reply:
That makes sense, but it might be confusing for reviewers.